File: Data Structures & Algorithms/valid-sudoku/submission-1.js
class Solution {
    /**
     * @param {character[][]} board
     * @return {boolean}
     */
    isValidSudoku(board) {
        // one pass to check row condition
        // one pass to check col condition
        // one pass to check 3 * 3 matrix condition 
        for(let i = 0; i<board.length; i++){
            const row_set = new Set();
            const col_set = new Set();
            for(let j = 0; j<board[0].length; j++){
                const row_el = board[i][j];
                if(row_el !== "."){
                if(row_set.has(row_el)){
                    return false;
                }else{
                    row_set.add(row_el)
                }
                }

                const col_el = board[j][i];
                if(col_el !== "." ){
                if(col_set.has(col_el)){
                    return false;
                }else{
                    col_set.add(col_el);
                }
                }
            }
        }

        

        for (let i = 0; i < 9; i += 3) {
    for (let j = 0; j < 9; j += 3) {

        const blockSet = new Set(); 

        for (let l = i; l < i + 3; l++) {
            for (let m = j; m < j + 3; m++) {

                const val = board[l][m];

                if (val !== ".") {

                if (blockSet.has(val)) {
                    return false;
                }else{
                blockSet.add(val);
                }}
            }
        }
    }
}

        // I have to find a way to navigate the grid in a 3 * 3 way

        return true;

    }
}

File: Data Structures & Algorithms/max-area-of-island/submission-1.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {

        int row = grid.length;
        int col = grid[0].length;

        int maxArea = 0;

        for(int i = 0; i<row; i++){
            for ( int j = 0; j<col; j++){
                if(grid[i][j] == 1){
                  int  area = dfs(grid, i, j);
                    maxArea  = Math.max(maxArea, area);
                }
            }
        }

        return maxArea;
        
    }

    public int dfs(int[][] grid, int r, int c){

        int row = grid.length;
        int col = grid[0].length;

        if(r < 0 || c < 0 || r >=row || c >=col){
            return 0;
        }

        if(grid[r][c] == 0){
            return 0;
        }

        grid[r][c] = 0;

        int area = 1;

       area += dfs(grid, r - 1, c);
       area += dfs(grid, r + 1, c);
       area += dfs(grid, r, c - 1);
       area += dfs(grid, r, c + 1);

       return area;
    
    }
}
